static/app/components/discover/quickContextCommitRow.tsx

```diff
@@ -0,0 +1,94 @@
+import styled from '@emotion/styled';
+
+import UserAvatar from 'sentry/components/avatar/userAvatar';
+import CommitLink from 'sentry/components/commitLink';
+import {PanelItem} from 'sentry/components/panels';
+import TextOverflow from 'sentry/components/textOverflow';
+import {t, tct} from 'sentry/locale';
+import ConfigStore from 'sentry/stores/configStore';
+import space from 'sentry/styles/space';
+import {Commit} from 'sentry/types';
+
+import {formatCommitMessage} from '../commitRow';
+import ExternalLink from '../links/externalLink';
+
+interface QuickContextCommitRowProps {
+  commit: Commit;
+  customAvatar?: React.ReactNode;
+}
+
+function QuickContextCommitRow({commit, customAvatar}: QuickContextCommitRowProps) {
+  const user = ConfigStore.get('user');
+  const isUser = user?.id === commit.author?.id;
+  const hasPullRequestURL = commit.pullRequest && commit.pullRequest.externalUrl;
+
+  return (
+    <StyledPanelItem key={commit.id} data-test-id="quick-context-commit-row">
+      {customAvatar
+        ? customAvatar
+        : commit.author && (
+            <div>
+              <UserAvatar size={24} user={commit.author} />
+            </div>
+          )}
+
+      <CommitLinks>
+        {hasPullRequestURL && commit.message && (
+          <LinkToPullRequest data-test-id="quick-context-commit-row-pr-link">
+            <ExternalLink href={commit.pullRequest?.externalUrl}>
+              {formatCommitMessage(commit.message)}
+            </ExternalLink>
+          </LinkToPullRequest>
+        )}
+        <LinkToCommit
+          prTitle={hasPullRequestURL && commit.message}
+          data-test-id="quick-context-commit-row-commit-link"
+        >
+          {tct('View commit [commitLink] by [author]', {
+            author: isUser ? t('You') : commit.author?.name ?? t('Unknown author'),
+            commitLink: (
+              <CommitLink
+                inline
+                showIcon={false}
+                commitId={commit.id}
+                repository={commit.repository}
+              />
+            ),
+          })}
+        </LinkToCommit>
+      </CommitLinks>
+    </StyledPanelItem>
+  );
+}
+
+const StyledPanelItem = styled(PanelItem)`
+  display: flex;
+  align-items: center;
+  gap: ${space(1)};
+  padding: 0;
+  border: none;
+
+  & + & {
+    margin-top: ${space(1)};
+  }
```

Comment:
Do we ever have two components next to each other here? You already have `gap` on line 67 and this is a flexbox, so wouldn't the spacing work itself out?

Reply:
Hey, that gap is for
<img width="336" alt="Screen Shot 2022-10-24 at 5 24 08 PM" src="https://user-images.githubusercontent.com/60121741/197632559-4bfb8eb3-9f6f-4bed-8462-e9f408806b5e.png">

We can have multiple commit rows.
So that & + & style works like this:
<img width="412" alt="Screen Shot 2022-10-24 at 5 28 00 PM" src="https://user-images.githubusercontent.com/60121741/197633056-cb41fbc9-182d-4f52-add4-5128c4edf6ad.png">